src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -1539,36 +1546,41 @@ private static List<String> removeEmptyCompileSourceRoots(List<String> compileSo
      * generated classes and if we got a file which is &gt;= the build-started timestamp, then we caught a file which
      * got changed during this build.
      *
-     * @return <code>true</code> if at least one single dependency has changed.
+     * @return {@code true} if at least one single dependency has changed.
      */
-    protected boolean isDependencyChanged() {
-        if (session == null) {
+    private boolean isDependencyChanged() {
```

Comment:
breaking backward compat protected -> private

Reply:
Same, this should have never been protected, is an internal method.

A good practice is to always start with the most restricted modifier (private) and then if needed open it to package-protected, then protected and finally public.

Again, this should not break anything since this class is not intended to be subclassed externally (unless I'm missing a weird use case and maven plugins can be used as public APIs).

Reply:
same comment as here https://github.com/apache/maven-compiler-plugin/pull/181#discussion_r1404816887